src/client/store/ui/geo/actions/postLayer.ts

```diff
@@ -1,27 +1,27 @@
 import { createAsyncThunk } from '@reduxjs/toolkit'
 import axios from 'axios'
 
+import { LayerResponseData } from '@meta/api/request/geo/layer'
 import { CountryIso } from '@meta/area'
 import { LayerKey, LayerSectionKey, sectionsApiEndpoint } from '@meta/geo'
 
-import { LayerState } from '../stateType'
+import { LayersSectionState, LayerState } from '../stateType'
 import { _getLayerRequestBody } from '.'
 
 export interface PostLayerProps {
   countryIso: CountryIso
   sectionKey: LayerSectionKey
   layerKey: LayerKey
   layerState: LayerState
+  sectionState?: LayersSectionState
 }
 
-export const postLayer = createAsyncThunk<[LayerSectionKey, LayerKey, string], PostLayerProps>(
+export const postLayer = createAsyncThunk<[LayerSectionKey, LayerKey, LayerResponseData], PostLayerProps>(
   'geo/post/layer',
-  async ({ countryIso, sectionKey, layerKey, layerState }) => {
-    const body = _getLayerRequestBody(countryIso, layerKey, layerState)
+  async ({ countryIso, sectionKey, layerKey, layerState, sectionState }) => {
```

Comment:
Better not to pass the state . Here you have access to getState directly .

Reply:
Good one 👍 Thanks!